clothcalc_placeholder.js

```diff
@@ -4374,26 +4376,17 @@
                       var str = $.trim($(this).children('span:last-child').html());
                       str = str.split('&nbsp;').join(' ');
                       switch (index) {
-                        case 0 :    data.motivation = parseInt(str.slice(0,str.indexOf(' ')), 10);
-                                  break;
-                        case 1 :    /** TODO: remove that evil eval **/
-                                  var tmp = str.replace('h',' * 3600 + ');
-                                  tmp = tmp.replace('m',' * 60 + ');
-                                  tmp = tmp.replace('s',' * 1 + ');
-                                  tmp += '0';
-                                  try {
-                                    data.duration = parseInt(eval(tmp), 10);
-                                  }
-                                  catch (e) {
-                                    throw {message:"unrecognized time on report: " + str};
-                                  }
-                                  break;
-                        case 2 :    data.wage = parseInt(str.slice(str.indexOf(' ')+1), 10);
-                                  break;
-                        case 3 :    data.bond = parseInt(str, 10);
-                                  break;
-                        case 4 :    data.experience = parseInt(str.slice(0,str.indexOf(' ')), 10);
-                                  break;
+                        case 0 : data.motivation = parseInt(str.slice(0,str.indexOf(' ')), 10);
+                                break;
+                        case 1 : var temp = parseFloat(str);
+                                 data.duration = temp == 1 ? 3600 : temp == 10 ? 600 : temp;
```

Comment:
How sure are you this will cover all the cases? What will this do on speed worlds, if such a thing even still exists? :)